src/lib/sw.js

```diff
@@ -117,29 +93,21 @@ workboxRouting.registerRoute(
 // Configure default cache for standard web.dev files: the offline page, various images etc.
 precacheAndRoute(manifest);
 
-/**
- * Match "/foo-bar/ "and "/foo-bar/as/many/of-these-as-you-like/" (with optional trailing
- * "index.html"), normal page nodes for web.dev. This only matches on pathname.
- */
-const contentPathRe = new RegExp("^/([\\w-]+/)*(|index.html)$");
-
 /**
  * Match "/foo-bar" and "/foo-bar/as-many/but/no/trailing/slash" (but not "/foo/bar/index.html").
  * This only matches on pathname (so it must always start with "/").
  */
 const untrailedContentPathRe = new RegExp("^(/[\\w-]+)+$");
 
 /**
- * Send normal nodes to cache first.
+ * Match fetches for patials, for SPA requests. Matches "/foo-bar/index.json" and
+ * "/foo-bar/many/parts/index.json", for partial SPA requests.
  */
-workboxRouting.registerRoute(
-  ({url, event}) => {
-    return url.host === self.location.host && contentPathRe.test(url.pathname);
-  },
-  new workboxStrategies.NetworkFirst({
-    plugins: [normalizeIndexCacheKeyPlugin],
-  }),
-);
+const partialPathRe = new RegExp("^/([\\w-]+/)*index\\.json$");
+const partialStrategy = new workboxStrategies.NetworkFirst();
+workboxRouting.registerRoute(({url, event}) => {
```

Comment:
Do you want to package up the same-origin check + pathname RegExp check into a helper function that generates a [`matchCallback`](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-routing#~matchCallback), since you use it a lot?

E.g.

```
function matchSameOriginRegExp(pathRegexp) {
  return ({url}) => url.host === self.location.host && pathRegexp.test(url.pathname);
}
```

and then you could write

```
workboxRouting.registerRoute(
  matchSameOriginRegExp(partialPathRe),
  partialStrategy
);
```

etc. throughout the service worker.

Reply:
Good call. Done.